### PR TITLE
feat: [SRTP-1980] add GET /activations/payer/status endpoint and tests

### PR DESCRIPTION
**Pull request by michele-fucito**
### Description

This PR adds a functional test suite for the `GET /activations/payer/{payerId}/status` endpoint introduced in the `rtp-activator` service. The endpoint returns a minimal `{ "isActive": true/false }` response indicating whether a payer (identified by fiscal code or VAT number) is currently enrolled in the RTP system, without exposing any PII or cross-SP information.

### List of changes

- **`functional-tests/tests/activation/test_debtor_activation_payer_status.py`** — new test suite (12 test cases, 32 parametrized variants) covering:
  - Happy paths: active payer same SP, never-activated payer, payer active on a different SP (privacy rule enforcement), `read_rtp_all` admin role, deactivated payer, omocodia fiscal code, foreign fiscal code, VAT number (Partita IVA)
  - Unhappy paths: invalid token (401), 22 parametrized invalid `payerId` formats (400) mirroring the Java test data source and extended with CF/PIVA structural edge cases
- **`api/debtor_activation_api.py`** — new client function `get_activation_status_by_fiscal_code`; `payerId` is passed as a **path parameter** (`/activations/payer/{payerId}/status`)
- **`api/utils/endpoints.py`** — new `ACTIVATION_PAYER_STATUS_URL` template constant
- **`config.yaml`** — new `activation_payer_status_path: "/activations/payer/{payerId}/status"` entry
- **`utils/dataset_payer_id_invalid.py`** — new shared dataset of invalid `payerId` values, reusable across test modules

### Motivation and context

The `GET /activations/payer/{payerId}/status` endpoint was added to allow Service Providers to check whether a payer is enrolled in RTP without fetching full activation records. A key privacy rule governs it: the response is deliberately indistinguishable between "payer not found" and "payer active under a different SP" — both return `{ "isActive": false }` — to prevent SPs from probing competitors' customer bases. This suite validates both the business logic and the privacy guarantee across RBAC roles (`read_rtp_activations` vs `read_rtp_all`).

> **Note:** the `cf_wrong_checksum` entry in the parametrized 400 suite is intentionally kept as a TDD sentinel — it will stay red until the backend enforces Italian CF checksum validation.

### Aggregation level

- [ ] Test case
- [x] Test suite

### Test type

- [x] Functional test
- [ ] Bdd test
- [ ] Performance
- [ ] End to end

### Type of changes

- [x] Add new test case/suite
- [ ] Modify existing test case/suite

### Test Results

- [ ] Success
- [x] Failure
- [ ] Poor performance
- [ ] Good performance

### Did you update secrets accordingly?

- [x] Not needed

### Did you update dependencies accordingly?

- [x] Not needed

### CODE REVIEW

File: api/debtor_activation_api.py [hunk 1]
```diff
@@ -59,6 +65,45 @@ def get_activation_by_id(access_token: str, activation_id: str):
     )
 
 
+def get_activation_status_by_fiscal_code(access_token: str, payer_fiscal_code: str):
+    """API to check whether a payer is active, returning minimal yes/no without PII.
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply api/debtor_activation_api.py:69]: the name covers the actual implementation.

File: functional-tests/tests/activation/test_debtor_activation_payer_status.py [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    ("invalid_birth_day_72_female", "XCGCHS98M72F166E"),
+    ("non_alphanumeric_character", "XCGCHS98M13F166!"),
+    ("space_in_fiscal_code", "XCGCHS98M13 F66E"),
+    ("cf_with_leading_whitespace", " XCGCHS98M13F166E"),
+    ("cf_with_trailing_whitespace", "XCGCHS98M13F166E "),
+    ("non_numeric_vat_like", "ABCDEFGHIJK"),
+    ("all_zeros_10_digits", "0000000000"),
+    ("vat_with_eu_it_prefix", "IT12345678903"),
+    ("cf_wrong_checksum", "RSSMRA85T10A562T"),
+    ("empty_string", ""),
```
[the remark this answers is not in the document]
Comment (michele-fucito) [reply functional-tests/tests/activation/test_debtor_activation_payer_status.py]: it's working as intended since we are working in TDD.

File: config.yaml [hunk 3]
```diff
@@ -91,6 +91,7 @@ iccrea_send_url: 'https://r5i-svil.api.gbi.bcc.it/pagopa4/pagopa-broker-api/pago
 # ACTIVATION LIST AND DETAILS
 # ============================
 activation_list_path: "/activations"
+activation_payer_status_path: "/activations/payer/status"
```
Comment (fmuscianisi) [inline config.yaml]: We should note that in this URL the `payerId` _path param_ is missing.
See the relevant [Design Review](https://pagopa.atlassian.net/wiki/spaces/RTP/pages/1285062880/Design+Review+-+RTP#Flusso-di-verifica-dello-stato-di-attivazione-di-un-cittadino) for reference.

File: README.md [hunk 4]
```diff
@@ -93,7 +93,7 @@ pytest functional-tests/tests/ -q
 
 | Directory | Coverage |
 |-----------|----------|
-| `activation/` | Debtor activation create, get, list, deactivation, security |
+| `activation/` | Debtor activation create, get, list, deactivation, payer status (GET /activations/payer/status), security |
```
Comment (fmuscianisi) [inline README.md]: See [this comment](https://github.com/pagopa/rtp-platform-qa/pull/190/changes#r3466009150).

File: api/debtor_activation_api.py [hunk 5]
```diff
@@ -59,6 +65,45 @@ def get_activation_by_id(access_token: str, activation_id: str):
...
+            "Authorization": f"{access_token}",
+            "Version": ACTIVATION_VERSION,
+            "RequestId": str(uuid.uuid4()),
+            "PayerId": payer_fiscal_code,
+        },
+        timeout=HTTP_TIMEOUT,
+    )
+
+
+def get_activation_status_without_payer_id(access_token: str):
```
Comment (fmuscianisi) [inline api/debtor_activation_api.py]: This case should not be possible in practice because the `payerId` must be passed as _path param_.
See [this comment](https://github.com/pagopa/rtp-platform-qa/pull/190/changes#r3466009150) for reference.

File: functional-tests/tests/activation/test_debtor_activation_payer_status.py [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+  as a TDD sentinel: it stays red until the backend enforces checksum validation.
+"""
+
+import allure
+import pytest
+
+from api.debtor_activation_api import get_activation_status_by_fiscal_code, get_activation_status_without_payer_id
+from api.debtor_deactivation_api import deactivate
+
+_INVALID_PAYER_IDS = [
```
Comment (fmuscianisi) [inline functional-tests/tests/activation/test_debtor_activation_payer_status.py]: Maybe these values can be abstracted into a config of its own.
This way other tests requiring assertions on _fiscal codes_ can benefit from them.

**Review by fmuscianisi — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 3 above)
Comment (michele-fucito) [reply config.yaml]: solved

File: README.md (hunk 4 above)
Comment (michele-fucito) [reply README.md]: solved

File: api/debtor_activation_api.py (hunk 5 above)
Comment (michele-fucito) [reply api/debtor_activation_api.py]: solved

File: functional-tests/tests/activation/test_debtor_activation_payer_status.py (hunk 6 above)
Comment (michele-fucito) [reply functional-tests/tests/activation/test_debtor_activation_payer_status.py]: solved

**Review by fmuscianisi — APPROVED** [review]
